### PR TITLE
Add github workflow to build the code

### PR DESCRIPTION
**Pull request by csbnw**
**Description**

As proposed in #197, the code is now actually being build in the CI by adding a new workflow to do so.

**Related issues**:

- https://github.com/nlesc-recruit/cudawrappers/issues/197

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  workflow_dispatch:
+
+jobs:
+  linting:
+    name: run
+    runs-on: ubuntu-latest
+    env:
+        # set tools versions to latest supported by the CI runner,
+        # ubuntu 22.04 at the time of writing
+        gcc-version: "12"  # = 12.1.0
```
Comment (matmanc) [inline .github/workflows/build.yml:14]: this is actually 12.3.0 at the time of running.

Maybe just remove the comment as it is misleading

**Review by matmanc — APPROVED** [review]
Looks good except fo the comment

File: .github/workflows/build.yml (hunk 1 above)
Comment (csbnw) [inline .github/workflows/build.yml:14]:
```suggestion
        gcc-version: "12" 
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  push:
+  workflow_dispatch:
+
+jobs:
+  linting:
+    name: run
+    runs-on: ubuntu-latest
+    env:
+        # set tools versions to latest supported by the CI runner,
+        # ubuntu 22.04 at the time of writing
```
Comment (csbnw) [inline .github/workflows/build.yml:13]: This comment is not really helpful and can be removed:
```suggestion
        # set tools versions to latest supported by the CI runner,
        # ubuntu 22.04 at the time of writing
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        gcc-version: "12" 
+        # cuda version to install with separate action
```
Comment (csbnw) [inline .github/workflows/build.yml:15]: The code is fairly obvious, so the comment is redudant.
```suggestion
        # cuda version to install with separate action
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (csbnw) [reply .github/workflows/build.yml]: You are right. The comment is now removed. I also removed the other comments, as they are not very helpfull either.

**Pull request merged**